modules/mono/editor/bindings_generator.cpp

```diff
@@ -678,7 +678,7 @@ void BindingsGenerator::_append_xml_constant(StringBuilder &p_xml_output, const
 				p_xml_output.append("<see cref=\"" BINDINGS_NAMESPACE ".");
 				p_xml_output.append(p_target_itype->proxy_name);
 				p_xml_output.append(".");
-				p_xml_output.append(target_ienum->cname);
+				p_xml_output.append(pascal_to_pascal_case(target_ienum->cname));
```

Comment:
Interesting, I'm pretty sure the `EnumInterface::cname` is already the correct name, as we set it in the `_populate_object_type_interfaces` when creating the `EnumInterface`.

https://github.com/godotengine/godot/blob/a7583881af5477cd73110cc859fecf7ceaf39bd7/modules/mono/editor/bindings_generator.cpp#L3217
https://github.com/godotengine/godot/blob/a7583881af5477cd73110cc859fecf7ceaf39bd7/modules/mono/editor/bindings_generator.cpp#L3225

Can you give an example where the issue happens?

Reply:
Off the top of my head, `Godot.MIDIMessage.<enum>` will still happen, even when it's correctly changed to `Godot.MidiMessage` when no enum type is specified

Reply:
Ah, I missed that we're only assigning the changed name if the enum name matches an existing type, so only when adding the `Enum` suffix. And it seems we are already using `pascal_to_pascal_case` every time we append the enum name, so your change seems fine. It does seem a bit odd that other `*Interface` types have a `proxy_name` field but `EnumInterface` doesn't.

Reply:
`EnumInterface` could probably stand to have `proxy_name` as well; it'd keep things consistent with the other structs & would eliminate edgecases like this from happening in the first place

Reply:
Implemented just that! Now `EnumInterface` has the variable `proxy_name` & its constructor was updated to mirror `ConstantInterface` by taking in three arguments (name, proxy_name, is_flag). The existing enum conversions to `pascal_to_pascal_case` now use `proxy_name` instead